classes/Customer.php

```diff
@@ -338,6 +342,7 @@ public function getByEmail($email, $passwd = null, $ignore_guest = true)
                 $this->{$key} = $value;
             }
         }
+        $this->id_default_group = (int)Configuration::get('PS_CUSTOMER_GROUP');
```

Comment:
@LittleBigDev I couldn't find any other solution, so in the method `Customer::getByEmail()` I change the `"id_default_group"`, like that it will pervert the problem that could occurs in the case as you state.